crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -166,6 +168,34 @@ impl<'w, 's, 'a> ChildBuilder<'w, 's, 'a> {
 pub trait BuildChildren {
     /// Creates a [`ChildBuilder`] with the given children built in the given closure
     fn with_children(&mut self, f: impl FnOnce(&mut ChildBuilder)) -> &mut Self;
+    /// Creates a [`ChildBuilder`] with the given children built in the given closure
+    ///
+    /// Compared to [`with_children`][BuildChildren::with_children], this method lets you
+    /// return data out of the closure like so:
+    ///
+    /// ```
+    /// # use bevy_ecs::prelude::*;
+    /// # use bevy_hierarchy::*;
+    /// #
+    /// # #[derive(Component)]
+    /// # struct SomethingElse;
+    /// #
+    /// # #[derive(Component)]
+    /// # struct MoreStuff;
+    /// #
+    /// # fn foo(mut commands: Commands) {
```

Comment:
There's little point in this doctest being executed, since `foo` is never called. That is, the doc test is merely testing that the function with no statements doesn't panic, which is trivially true.

Alternatively, we could create a world and SystemState of commands, and actually test that running the command works.

However, `add_children` cannot panic, so I'm not really sure I see the point.

Reply:
Well does cargo test still compile it if it's `no_run`?

Reply:
Ah, just checked and the answer is yes. Will mark as no_run.